osutil/mountentry.go

```diff
@@ -132,6 +132,20 @@ func (e *MountEntry) OptStr(name string) (string, bool) {
 	return "", false
 }
 
+// OptStrMod modifies and existing key=value mount option while
+// preserving the option order.
+func (e *MountEntry) OptStrMod(name string, value string) {
```

Comment:
as I said elsewhere it might be slightly preferable not mutating things in place but given that we do, my preference would be an API like:

osutil.ReplaceMountEntryOption(mountEntry, XSnapdMustExistDir(newPath))  

instead

Reply:
This is done and in use in the replacement: https://github.com/snapcore/snapd/pull/13342/files